pyspeckit/spectrum/models/ammonia.py

```diff
@@ -179,41 +181,43 @@ def ammonia(xarr, tkin=20, tex=None, ntot=14, width=1, xoff_v=0.0,
                                        (Crot-Brot)*Jpara**2)/(kb*tkin))
         Zortho = 2*(2*Jortho+1)*np.exp(-h*(Brot*Jortho*(Jortho+1)+
                                            (Crot-Brot)*Jortho**2)/(kb*tkin))
+        Qpara = Zpara.sum()
+        Qortho = Zortho.sum()
         for linename in line_names:
             if ortho_dict[linename]:
                 orthoparafrac = fortho
                 Z = Zortho
+                Qtot = Qortho
                 count = ortho_count
                 ortho_count += 1
             else:
                 orthoparafrac = 1.0-fortho
                 Z = Zpara
+                Qtot = Qpara
                 count = para_count # need to treat partition function separately
                 para_count += 1
 
+
+            # for a complete discussion of these equations, please see
+            # https://github.com/keflavich/pyspeckit/blob/ammonia_equations/examples/AmmoniaLevelPopulation.ipynb
+            # and
+            # http://low-sky.github.io/ammoniacolumn/
+
             # short variable names for readability
             frq = freq_dict[linename]
             partition = Z[count]
             aval = aval_dict[linename]
 
-            # Friesen 2009 eqn A4 points out that the partition function actually says
-            # how many molecules are in the NH3(1-1) state, both upper *and* lower.
-            # population_upperlower = ntot * orthoparafrac * partition/(Z.sum())
-            # population_upperstate = population_upperlower / (1+np.exp(h*frq/(kb*tex)))
-            #
-            # Note Jan 1, 2015: This is accounted for in the eqn below.  The
-            # only difference is that I have used Tkin where Friesen et al 2009
-            # use Tex.  Since Tex describes which states are populated, that may
-            # be the correct one to use.
-
             # Total population of the higher energy inversion transition
-            population_upperstate = lin_ntot * orthoparafrac * partition/(Z.sum())
+            population_upperstate = lin_ntot * orthoparafrac * partition/Qtot
```

Comment:
I tried to amend this in the notebook [here](https://github.com/low-sky/pyspeckit/tree/pr136), but I'm always suspicious I have a deep misunderstanding of stat mech...

I am arguing for `N(1,1) = N(1,1)_a + N(1,1)_s` for symmetric/antisymmetric states respectively.  The energy used here in Boltzmann is the energy for the rotation of the J,J state.  I think the full energies should be:
`E(J,J)_a = h*(Brot*(J*(J+1)) - (Crot - Brot)*J**2)) + h*nu0/2`
`E(J,J)_s = h*(Brot*(J*(J+1)) - (Crot - Brot)*J**2)) - h*nu0/2`
where I've cheated and used `h*nu0` as the energy difference between the inversion states.   And, of course, this says nothing of the hyperfine splitting but we assume that the energies of the different HFS-producing levels are all similar so we can decouple that part of the problem into the weights of the different hyperfine components depending only on the statistical weights rather than the energy differences.